Vizzuality/marxan-cloud
# get admin areas by country

### Overview

New or updated endpoints:

`GET /api/v1/countries/:id/administrative-areas` should list (with pagination) admin areas within a given country, with `:id` being the ISO3166 alpha-3 code for the country (uppercase).

`GET /api/v1/administrative-areas` should list (with pagination) admin areas.

`GET /api/v1/administrative-areas/:id` should return the level 1 or level 2 admin area matching the id provided.

MARXAN-135 includes discussions of more advanced filters for searching admin areas within a country, but I will do that after we define the requirements for these filters, in a separate PR. This can be reviewed as is.

### Designs

N/A

### Testing instructions

See description above.

We do have some e2e/contract tests in this PR, but they will fail in CI until we change the GitHub actions flow to include seeding the geodb with a minimal set of gadm data.

Running `docker-compose exec api yarn test:e2e` locally should work, however, if the local instance has GADM data.

### Feature relevant tickets

https://vizzuality.atlassian.net/browse/MARXAN-135
https://vizzuality.atlassian.net/browse/MARXAN-134

---

## Checklist before submitting

      deploying to staging/production, please add brief testing instructions
      to the deploy checklist (`docs/deployment-checklist.md`)
